test/provider_os_memory_multiple_numa_nodes.cpp

```diff
@@ -425,6 +425,165 @@ TEST_F(testNuma, checkModeInterleaveCustomPartSize) {
+                                numaSplitOut(4, {1}),
+                                {{2}},
+                                numaSplitOut(10, {1}),
+                                {{2}},
+                                numaSplitOut(17, {0})});
+
+        ret.push_back({3, 35, in, out});
+
+        out = mergeOutVec({{{0}, {0, 1}},
+                           numaSplitOut(3, {1}),
+                           {{1, 2}, {1, 2}},
+                           numaSplitOut(8, {1}),
+                           {{0, 1, 2}},
+                           numaSplitOut(15, {0})});
+
+        ret.push_back({3, 31, in, out});
+
+        in = {{31, 1}, {17, 1}, {14, 2}, {19, 0}, {8, 2}, {5, 1}, {15, 2}};
+        out = mergeOutVec({
+            numaSplitOut(28, {1}),
+            {{1, 2}},
+            numaSplitOut(7, {2}),
+            {{0, 2}},
+            numaSplitOut(11, {0}),
+            {{0, 2}},
+            numaSplitOut(4, {2}),
+            {{1, 2}},
+            numaSplitOut(2, {1}),
+            {{1, 2}},
+            numaSplitOut(8, {2}),
+        });
+
+        ret.push_back({3, 65, in, out});
+
+        out = {{0, 1, 2}};
+        ret.push_back({3, 1, in, out});
+
+        out = {{1}, {0, 1, 2}, {0, 1, 2}};
+        ret.push_back({3, 3, in, out});
+
+        in = {};
+        out = {{0}, {0, 1}, {1}, {1, 2}, {2}};
+        ret.push_back({3, 5, in, out});
+        out = {{0}, {0}, {0, 1}, {1}, {1}};
+        ret.push_back({2, 5, in, out});
+
+        std::vector<unsigned> numa_nodes = get_available_numa_nodes();
+
+        std::vector<umf_numa_split_partition_t> in1, in2, in3;
+
+        for (unsigned i = 0; i < numa_nodes.size(); i++) {
+            numaSplitOut out1, out2, out3;
+            std::vector<umf_numa_split_partition_t> in1, in2, in3;
+            in1 = {};
+            out1 = {{}};
+            for (unsigned j = 0; j <= i; j++) {
+                in2.push_back({42, j});
+                in3.push_back({j + 1, j});
+                out1[0].push_back(j);
+                out2.push_back({j});
+                out3 = mergeOutVec({out3, numaSplitOut(j + 1, {j})});
+            }
+            unsigned z = i + 1;
+            ret.push_back({z, 1, in1, out1});
+            ret.push_back({z, z, in2, out2});
+            ret.push_back({z, z * (z - 1) / 2 + z, in3, out3});
+        }
+        return ret;
+    }
+};
+
+INSTANTIATE_TEST_SUITE_P(checkModeSplit, testNumaSplit,
+                         ::testing::ValuesIn(testNumaSplit::getTestInput()));
+
+TEST_P(testNumaSplit, checkModeSplit) {
+    auto in = GetParam();
+    size_t page_size = sysconf(_SC_PAGE_SIZE);
+
+    umf_os_memory_provider_params_t os_memory_provider_params =
+        UMF_OS_MEMORY_PROVIDER_PARAMS_TEST;
+
+    std::vector<unsigned> numa_nodes = get_available_numa_nodes();
+
+    if (numa_nodes.size() < std::get<0>(in)) {
+        GTEST_SKIP_("Not enought numa nodes");
+    }
+
+    ASSERT_EQ(std::get<3>(in).size(), std::get<1>(in)) << "Wrong test in data";
```

Comment:
It was sanity check to ensure that i defined correctly test input

Reply:
sooo... it should be removed...?

Reply:
It might - but i find it useful if someone wants to extend test.
It is not easy to create output array, and it is not easy to find out if there is just entry missing

Reply:
now it looks better, thanks for defining named params instead of these `get<>` calls.

The question still stands - what `Wrong test in data` mean here...? did you mean `wrong test's params` or, more precise `out array size does not match pages count`...?

Reply:
Done